core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -102,9 +102,17 @@ public Schema record(Schema record, List<String> names, Iterable<Schema.Field> s
 
       } else {
         Preconditions.checkArgument(
-            field.isOptional() || MetadataColumns.metadataFieldIds().contains(field.fieldId()),
-            "Missing required field: %s",
+            (field.isRequired() && field.initialDefault() != null)
+                || field.isOptional()
+                || MetadataColumns.metadataFieldIds().contains(field.fieldId()),
+            "Missing required field that doesn't have a default value: %s",
             field.name());
+        // If the field from Iceberg schema has initial default value, we just pass and don't
```

Comment:
Style: In Iceberg, we don't use personal pronouns like "I" or "we" in documentation or comments because it makes documentation harder to understand (who is "we"?) and longer. Instead, be direct:

> If the field has an initial default value, do not project it in the read schema. The default will be added in the Iceberg `ValueReader`.